File: SmartStayHaven.API/Models/Charge.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartStayHaven.API.Models
{
    public class Charge
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int BookingId { get; set; }

        [Required]
        public string Description { get; set; } = string.Empty;

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        [Required]
        public decimal Quantity { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal UnitPrice { get; set; }

        [Required]
        public string Category { get; set; } = string.Empty; // Room, Food, Service, Amenity, Other

        public string? Notes { get; set; }

        public bool IsPaid { get; set; }

        public DateTime ChargeDate { get; set; } = DateTime.UtcNow;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? LastUpdated { get; set; }

        // Navigation properties
        public Booking? Booking { get; set; }
    }
} 

File: src/types/charge.ts

export type ChargeCategory = 'RoomRate' | 'Food' | 'Beverage' | 'Service' | 'Tax' | 'Other';
export type ChargeStatus = 'Pending' | 'Paid' | 'Cancelled' | 'Refunded';

export interface Charge {
  id: number;
  bookingId: number;
  booking?: any;
  category: ChargeCategory;
  description: string;
  amount: number;
  status: ChargeStatus;
  date: string;
  notes?: string;
  createdAt: string;
  lastUpdated?: string;
}

export interface ChargeCreate {
  bookingId: number;
  category: ChargeCategory;
  description: string;
  amount: number;
  status?: ChargeStatus;
  date?: string;
  notes?: string;
}

export interface ChargeUpdate {
  category?: ChargeCategory;
  description?: string;
  amount?: number;
  status?: ChargeStatus;
  notes?: string;
}
